File: TrabajoAcademicoNet/SoftProgModelo/ProductoModelo/Producto.cs
using PUCP.Inf30.SoftProg.Modelo.BaseModelo;
using PUCP.Inf30.SoftProg.Modelo.ProductoModelo.ProductoUtils;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PUCP.Inf30.SoftProg.Modelo.ProductoModelo
{
    public class Producto : Registro
    {
        public String Nombre { get; set; }
        public String SKU { get; set; }
        public String Descripcion { get; set; }
        public double PrecioUnitario { get; set; }
        public double PrecioAlMayor { get; set; }
        public UnidadMedida MedidaAlMayor;
        public int StockDisponible { get; set; }
        public int StockMinimo { get; set; }
        public int StockMaximo { get; set; }
        public int IdCategoria { get; set; }

        public Producto()
        {

        }
        public Producto(String nombre, String SKU, String descripcion, double precioUnitario,
            double precioAlMayor, UnidadMedida medidaAlMayor, int stockDisponible, int stockMinimo, int stockMaximo, int idCategoria)
        {
            this.Nombre = nombre;
            this.SKU = SKU;
            this.Descripcion = descripcion;
            this.PrecioUnitario = precioUnitario;
            this.PrecioAlMayor = precioAlMayor;
            this.MedidaAlMayor = medidaAlMayor;
            this.StockDisponible = stockDisponible;
            this.StockMinimo = stockMinimo;
            this.StockMaximo = stockMaximo;
            this.IdCategoria = idCategoria;
        }

        public Producto(String nombre, String SKU, String descripcion, double precioUnitario, double precioAlMayor,
            UnidadMedida medidaAlMayor, int stockDisponible, int stockMinimo, int stockMaximo, int idCategoria, int id, bool activo) :
            base(id, activo)
        {
            this.Nombre = nombre;
            this.SKU = SKU;
            this.Descripcion = descripcion;
            this.PrecioUnitario = precioUnitario;
            this.PrecioAlMayor = precioAlMayor;
            this.MedidaAlMayor = medidaAlMayor;
            this.StockDisponible = stockDisponible;
            this.StockMinimo = stockMinimo;
            this.StockMaximo = stockMaximo;
            this.IdCategoria = idCategoria;
        }
        public String getMedidaAlMayorString()
        {
            String Cadena = "";

            switch (this.MedidaAlMayor)
            {
                case UnidadMedida.BOTELLA:
                    Cadena = "BOTELLA";
                    break;
                case UnidadMedida.CAJA:
                    Cadena = "CAJA";
                    break;
                case UnidadMedida.GALON:
                    Cadena = "GALON";
                    break;
                case UnidadMedida.KILOGRAMO:
                    Cadena = "KILOGRAMO";
                    break;
                case UnidadMedida.LITRO:
                    Cadena = "LITRO";
                    break;
                case UnidadMedida.PACK:
                    Cadena = "PACK";
                    break;
                case UnidadMedida.PAQUETE:
                    Cadena = "PAQUETE";
                    break;
                case UnidadMedida.SACO:
                    Cadena = "SACO";
                    break;
            }
            return Cadena;
        }

        public void setMedidaAlMayorString(String medida)
        {
            switch (medida)
            {
                case "BOTELLA":
                    this.MedidaAlMayor = UnidadMedida.BOTELLA;
                    break;
                case "CAJA":
                    this.MedidaAlMayor = UnidadMedida.CAJA;
                    break;
                case "GALON":
                    this.MedidaAlMayor = UnidadMedida.GALON;
                    break;
                case "KILOGRAMO":
                    this.MedidaAlMayor = UnidadMedida.KILOGRAMO;
                    break;
                case "LITRO":
                    this.MedidaAlMayor = UnidadMedida.LITRO;
                    break;
                case "PACK":
                    this.MedidaAlMayor = UnidadMedida.PACK;
                    break;
                case "PAQUETE":
                    this.MedidaAlMayor = UnidadMedida.PAQUETE;
                    break;
                case "SACO":
                    this.MedidaAlMayor = UnidadMedida.SACO;
                    break;
            }
        }

    }
}

File: SoftProgModelo/src/main/java/pe/edu/pucp/inf30/softprog/modelo/producto/Producto.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package pe.edu.pucp.inf30.softprog.modelo.producto;

import pe.edu.pucp.inf30.softprog.modelo.base.Registro;
import pe.edu.pucp.inf30.softprog.modelo.producto.utils.UnidadMedida;

/**
 *
 * @author Cristhian Horacio
 */
public class Producto extends Registro {

    private String nombre;
    private String SKU;
    private String descripcion;
    private double precioUnitario;
    private double precioAlMayor;
    private UnidadMedida medidaAlMayor;
    private int stockDisponible;
    private int stockMinimo;
    private int stockMaximo;
    private int idCategoria;
    private String marca;
    
    public Producto() {

    }

    public Producto(String nombre, String SKU, String descripcion, double precioUnitario, double precioAlMayor, UnidadMedida medidaAlMayor, int stockDisponible, int stockMinimo, int stockMaximo, int idCategoria) {
        this.nombre = nombre;
        this.SKU = SKU;
        this.descripcion = descripcion;
        this.precioUnitario = precioUnitario;
        this.precioAlMayor = precioAlMayor;
        this.medidaAlMayor = medidaAlMayor;
        this.stockDisponible = stockDisponible;
        this.stockMinimo = stockMinimo;
        this.stockMaximo = stockMaximo;
        this.idCategoria = idCategoria;
    }

    public Producto(String nombre, String SKU, String descripcion, double precioUnitario, double precioAlMayor, UnidadMedida medidaAlMayor, int stockDisponible, int stockMinimo, int stockMaximo, int idCategoria, int id, boolean activo) {
        super(id, activo);
        this.nombre = nombre;
        this.SKU = SKU;
        this.descripcion = descripcion;
        this.precioUnitario = precioUnitario;
        this.precioAlMayor = precioAlMayor;
        this.medidaAlMayor = medidaAlMayor;
        this.stockDisponible = stockDisponible;
        this.stockMinimo = stockMinimo;
        this.stockMaximo = stockMaximo;
        this.idCategoria = idCategoria;
    }

    

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getSKU() {
        return SKU;
    }

    public void setSKU(String SKU) {
        this.SKU = SKU;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    public double getPrecioUnitario() {
        return precioUnitario;
    }

    public void setPrecioUnitario(double precioUnitario) {
        this.precioUnitario = precioUnitario;
    }

    public double getPrecioAlMayor() {
        return precioAlMayor;
    }

    public void setPrecioAlMayor(double precioAlMayor) {
        this.precioAlMayor = precioAlMayor;
    }

    public UnidadMedida getMedidaAlMayor() {
        return medidaAlMayor;
    }

    public void setMedidaAlMayor(UnidadMedida medidaAlMayor) {
        this.medidaAlMayor = medidaAlMayor;
    }

    public int getStockDisponible() {
        return stockDisponible;
    }

    public void setStockDisponible(int stockDisponible) {
        this.stockDisponible = stockDisponible;
    }

    public int getStockMinimo() {
        return stockMinimo;
    }

    public void setStockMinimo(int stockMinimo) {
        this.stockMinimo = stockMinimo;
    }

    public int getStockMaximo() {
        return stockMaximo;
    }

    public void setStockMaximo(int stockMaximo) {
        this.stockMaximo = stockMaximo;
    }

    public String getMedidaAlMayorString() {
        String cadena = "";

        switch (this.medidaAlMayor) {
            case BOTELLA ->
                cadena = "BOTELLA";
            case CAJA ->
                cadena = "CAJA";
            case GALON ->
                cadena = "GALON";
            case KILOGRAMO ->
                cadena = "KILOGRAMO";
            case LITRO ->
                cadena = "LITRO";
            case PACK ->
                cadena = "PACK";
            case PAQUETE ->
                cadena = "PAQUETE";
            case SACO ->
                cadena = "SACO";
        }
        return cadena;
    }

    public void setMedidaAlMayorString(String medida) {
        switch (medida) {
            case "BOTELLA" ->
                this.medidaAlMayor = UnidadMedida.BOTELLA;
            case "CAJA" ->
                this.medidaAlMayor = UnidadMedida.CAJA;
            case "GALON" ->
                this.medidaAlMayor = UnidadMedida.GALON;
            case "KILOGRAMO" ->
                this.medidaAlMayor = UnidadMedida.KILOGRAMO;
            case "LITRO" ->
                this.medidaAlMayor = UnidadMedida.LITRO;
            case "PACK" ->
                this.medidaAlMayor = UnidadMedida.PACK;
            case "PAQUETE" ->
                this.medidaAlMayor = UnidadMedida.PAQUETE;
            case "SACO" ->
                this.medidaAlMayor = UnidadMedida.SACO;
        }
    }

    public int getIdCategoria() {
        return idCategoria;
    }

    public void setIdCategoria(int idCategoria) {
        this.idCategoria = idCategoria;
    }


    public String getMarca() {
        return marca;
    }


    public void setMarca(String marca) {
        this.marca = marca;
    }
    
}
